docs/6-infrastructure-configuration-management/6.1.2-terraform-backends.md

```diff
@@ -0,0 +1,67 @@
+# Terraform State & Backends
+
+## Sharing workspaces
+
+In the last tutorial from the previous section you learned how to use Terraform Cloud to store a Terraform state remotely. Using remote states makes it possible for teams to share and create automation for Terraform workspaces. Work with your teammates to in the following exercises to get a better understand of how to share a Terraform workspace.
+
+1. Create a GitHub repository for your Terraform and push up your code. You should only include your `.tf` and `.terraform.lock.hcl`.
+
+2. Working with a teammate have them clone your repo, make a change to your Terraform and apply it and push the change up to your repo.
+
+?> Watch the state of the workspace in the Terraform Cloud as they apply the change
+
+3. Run `terraform plan` with your original configuration. Now pull down your teammates change and run `terraform plan` again and observe the difference.
+
+**Deep Dive**
+
+- What is the purpose of the `terraform.lock.hcl` file and why should it be included in the source repo?
+
+## Cloud native backends
+
+Terraform Cloud has a lot of features and is the foundation for Terraform Enterprise. But there are lots of other Terraform backends available for storing your state remotely. Lets take a look at using some cloud native storage solutions.
+
+If you ran through the AWS tutorial use the [S3 backend](https://developer.hashicorp.com/terraform/language/settings/backends/s3) for this exercise. For the Azure tutorial use the [azurerm backend](https://developer.hashicorp.com/terraform/language/settings/backends/azurerm).
+
+!> Terraform has not implemented state migration from Terraform Cloud to another backend so you will need to clean up the previous state with the following steps
+
+1. Run `terraform destroy` to remove existing resources.
+
+2. Delete the `.terraform` folder.
+
+3. Create cloud resources to store the Terraform state in. The specific resources you need will depend on the cloud platform and backend you are using. Read the documentation for the specific backend and Terraform resources for more information. 
+
+4. Remove the `cloud` configuration block from the `terraform` block of your Terraform configuration.
+
+5. Add a `backend` configuration block to the `terraform` block for your backend. The backend configuration should reference the resources you created in step #3.
+
+6. Run `terraform init` and `terraform apply` to initialize the new backend and apply the configuration.
+
+7. Look at the cloud storage you setup for the Terraform backend. What files have been created and what do they contain?
+
+8. Push up you changes and have a teammate pull down you changes and apply a change
```

Comment:
"Push up you[r] changes. Have a teammate pull down you[r] changes and apply a change"